packages/cli/tests/e2e/bot/ProvisionAppServiceNotificationBot.tests.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @author Xiaofu Huang <xiaofhua@microsoft.com>
+ */
+
+import fs from "fs-extra";
+import path from "path";
+
+import { AadValidator, BotValidator } from "../../commonlib";
+
+import {
+  execAsync,
+  execAsyncWithRetry,
+  getSubscriptionId,
+  getTestFolder,
+  getUniqueAppName,
+  cleanUp,
+  readContextMultiEnv,
+} from "../commonUtils";
+import AppStudioLogin from "../../../src/commonlib/appStudioLogin";
+import { environmentManager } from "@microsoft/teamsfx-core";
+
+import { it } from "../../commonlib/it";
+
+describe("Provision", function () {
+  const testFolder = getTestFolder();
+  const appName = getUniqueAppName();
+  const subscription = getSubscriptionId();
+  const projectPath = path.resolve(testFolder, appName);
+  const envName = environmentManager.getDefaultEnvName();
+
+  const env = Object.assign({}, process.env);
+  env["TEAMSFX_CONFIG_UNIFY"] = "true";
+  env["BOT_NOTIFICATION_ENABLED"] = "true";
+  env["TEAMSFX_TEMPLATE_PRERELEASE"] = "alpha";
+
+  it("Provision Resource: app service hosted notification", async function () {
+    await execAsync(
+      `teamsfx new --interactive false --app-name ${appName} --capabilities notification --bot-host-type-trigger http-restify --programming-language typescript`,
+      {
+        cwd: testFolder,
+        env: env,
+        timeout: 0,
+      }
+    );
+    console.log(`[Successfully] scaffold to ${projectPath}`);
+
+    // set subscription
+    await execAsync(`teamsfx account set --subscription ${subscription}`, {
+      cwd: projectPath,
+      env: env,
+      timeout: 0,
+    });
+
+    console.log(`[Successfully] set subscription for ${projectPath}`);
+
+    // provision
+    await execAsyncWithRetry(`teamsfx provision`, {
+      cwd: projectPath,
+      env: env,
+      timeout: 0,
+    });
+
+    console.log(`[Successfully] provision for ${projectPath}`);
+
+    {
+      // Validate provision
+      // Get context
+      const context = await readContextMultiEnv(projectPath, envName);
+
+      // Validate Aad App
+      const aad = AadValidator.init(context, false, AppStudioLogin);
+      await AadValidator.validate(aad);
+
+      // Validate Bot Provision
+      const bot = new BotValidator(context, projectPath, envName);
+      await bot.validateProvision();
+    }
+
+    // deploy
+    await execAsyncWithRetry(`teamsfx deploy bot`, {
```

Comment:
Just curious - It looks `teamsfx deploy bot` does not align with `teamsfx new` which specified capability as `notification`.

Reply:
Good point! @swatDong Do we need to change the deployment question?

Reply:
Agree. For better UX it's better to show `notification`, `command`, etc.
But may be hard to do in current framework since `bot` comes from plugin name.

Let's track in another work item.